articles/sorting-algorithms.md

```diff
@@ -0,0 +1,145 @@
+# Sorting algorithms visualised
+- 19th December 2016
+
+Sometimes you or a computer needs to sort a list of items—say, a list of names into alphabetical order, or a list of objects into order of length. To sort this list, especially if it is a large list, you need some kind of process: for example, if you were sorting a list of names into alphabetic order, you could put all the names beginning with A together, then all the Bs together, and then you could sort them by the second letter: all the AAs together, then all the ABs together, until they're sorted.
+
+Algorithms used to sort a list of items are called sorting algorithms, and in this article I'm going to explain and visualise a few of the most well known sorting algorithms. Each section will contain one sorting algorithm, a description of the algorithm, and then a visualisation of the algorithm, sorting different length bars (seen below) from shortest to longest, powered by [D3.js]. There are some controls to the bottom right of the screen: it might help if you adjust the speed of the visualisation a bit.
```

Comment:
"visualisation of the algorithm sorting different length bars from shortest to longest, as seen below. These are powered by [D3.js] and can be adjusted with the controls to the bottom right of the screen."